fvtest/compilertriltest/MinimalTest.cpp

```diff
@@ -0,0 +1,405 @@
+   {
+   Return(
+      Sub(
+         Load("arg1"),
+         Load("arg2")));
+   return true;
+   }
+
+class FactorialMethod : public StaticSignatureMethodBuilder<int32_t(int32_t)>
+   {
+   public:
+   FactorialMethod(TR::TypeDictionary *types);
+   virtual bool buildIL();
+   };
+
+FactorialMethod::FactorialMethod(TR::TypeDictionary *types)
+   : StaticSignatureMethodBuilder<int32_t(int32_t)>(types)
+   {
+   DefineLine(LINETOSTR(__LINE__));
+   DefineFile(__FILE__);
+
+   DefineName("Factorial");
+   DefineParameter("x", Int32);
+   DefineReturnType(Int32);
+   }
+
+bool
+FactorialMethod::buildIL()
+   {
+   TR::IlBuilder *baseCase=NULL, *recursiveCase=NULL;
+   IfThenElse(&baseCase, &recursiveCase,
+      LessThan(
+         Load("x"),
+         ConstInt32(1)));
+
+   DefineLocal("result", Int32);
+
+   baseCase->Store("result",
+   baseCase->   ConstInt32(1));
+   recursiveCase->Store("result",
+   recursiveCase->   Mul(
+   recursiveCase->      Call("Factorial", 1,
+   recursiveCase->         Sub(
+   recursiveCase->            Load("x"),
+   recursiveCase->            ConstInt32(1))),
+   recursiveCase->      Load("x")));
+
+   Return(
+      Load("result"));
+   return true;
+   }
+
+class RecursiveFibonnaciMethod : public StaticSignatureMethodBuilder<int32_t(int32_t)>
+   {
+   public:
+	RecursiveFibonnaciMethod(TR::TypeDictionary *types);
+   virtual bool buildIL();
+   };
+
+
+RecursiveFibonnaciMethod::RecursiveFibonnaciMethod(TR::TypeDictionary *types)
+   : StaticSignatureMethodBuilder<int32_t(int32_t)>(types)
+   {
+   DefineLine(LINETOSTR(__LINE__));
+   DefineFile(__FILE__);
+
+   DefineName("fib_recur");
+   DefineParameter("n", Int32);
+   DefineReturnType(Int32);
+   }
+
+bool
+RecursiveFibonnaciMethod::buildIL()
+   {
+   TR::IlBuilder *baseCase=NULL, *recursiveCase=NULL;
+   IfThenElse(&baseCase, &recursiveCase,
+      LessThan(
+         Load("n"),
+         ConstInt32(2)));
+
+   DefineLocal("result", Int32);
+
+   baseCase->Store("result",
+   baseCase->   Load("n"));
+   recursiveCase->Store("result",
+   recursiveCase->   Add(
+   recursiveCase->      Call("fib_recur", 1,
+   recursiveCase->         Sub(
+   recursiveCase->            Load("n"),
+   recursiveCase->            ConstInt32(1))),
+   recursiveCase->      Call("fib_recur", 1,
+   recursiveCase->         Sub(
+   recursiveCase->            Load("n"),
+   recursiveCase->            ConstInt32(2)))));
+
+   Return(
+      Load("result"));
+   return true;
+   }
+
+
+template <typename M>
+typename M::FunctionPtr compile()
+   {
+   TR::TypeDictionary types;
+   M m(&types);
+   return m.Compile();
+   }
+
+TEST_F(MinimalTest, MeaningOfLife)
+   {
+   std::string arch = omrsysinfo_get_CPU_architecture();
+   SKIP_IF((OMRPORT_ARCH_PPC == arch || OMRPORT_ARCH_PPC64 == arch || OMRPORT_ARCH_PPC64LE == arch), MissingImplementation)
+           << "Test is skipped on non-x86-64 platforms because calls are not currently supported on other POWER (see issue #1645)";
```

Comment:
I don't think the message is accurate since the test does run on some non-x86-64 platforms (and the issue referenced is not related).